src/compiler/objc_message.cc

```diff
@@ -1009,6 +1009,8 @@ namespace google { namespace protobuf { namespace compiler { namespace objective
           "  return NO;\n"
           "}\n",
           "capitalized_name", UnderscoresToCapitalizedCamelCase(field));
+      } else {
+        GenerateRequiredFieldCheckSource(printer, field);
```

Comment:
This code implies that something will be generated. Maybe worth restructuring to something like 
```
if (FiledHasRequiredComment()) {
    GenerateRequiredFieldCheckSource()
}
```
several methods or at least rename to  GenerateRequiredFieldCheckSourceIfNeeded()

Reply:
Good point! Replaced